plugins/inputs/t128_tank/reader.go

```diff
@@ -272,6 +272,10 @@ parseLoop:
 		messages, parseErr = parseLines(tankReader, r.topic)
 		if parseErr != nil {
 			r.log.Errorf("encountered error while parsing lines in %s output, will not attempt to parse more lines: %v", r.topic, parseErr)
+			if err := cmd.Process.Signal(syscall.SIGTERM); err != nil {
```

Comment:
It actually seems like we need a context within this function that is canceled when the function completes. Once canceled, we can wait. That would be much cleaner than signaling like this.